spec/spec_models.cr

```diff
@@ -0,0 +1,204 @@
+      table_name "{{ teacher_table }}"
+
+      field name : String
+
+      has_many :klass_{{ adapter_literal }}s
+
+      def self.drop_and_create
+        exec("DROP TABLE IF EXISTS {{ teacher_table }};")
+        exec("CREATE TABLE {{ teacher_table }} (
+          id {{ primary_key_sql }},
+          name VARCHAR(100)
+        );
+        ")
+      end
+    end
+
+    class Student{{ adapter_const_suffix }} < Granite::ORM::Base
+      primary id : Int64
+      adapter {{ adapter_literal }}
+      table_name "{{ student_table }}"
+
+      field name : String
+
+      has_many :enrollment_{{ adapter_literal }}s
+      has_many :klass_{{ adapter_literal }}s, through: :enrollment_{{ adapter_literal }}s
+
+      def self.drop_and_create
+        exec("DROP TABLE IF EXISTS {{ student_table }};")
+        exec("CREATE TABLE {{ student_table }} (
+          id {{ primary_key_sql }},
+          name VARCHAR(100),
+          parent{{ adapter_suffix }}_id {{ foreign_key_sql }}
+        );
+        ")
+      end
+    end
+
+    class Klass{{ adapter_const_suffix }} < Granite::ORM::Base
+      primary id : Int64
+      adapter {{ adapter_literal }}
+      table_name "{{ klass_table }}"
+      field name : String
+
+      belongs_to :teacher_{{ adapter_literal }}
+
+      has_many :enrollment_{{ adapter_literal }}s
+      has_many :student_{{ adapter_literal }}s, through: :enrollment_{{ adapter_literal }}s
+
+      def self.drop_and_create
+        exec "DROP TABLE IF EXISTS {{ klass_table }}"
+        exec <<-SQL
+          CREATE TABLE {{ klass_table }} (
+            id {{ primary_key_sql }},
+            name VARCHAR(255),
+            teacher{{ adapter_suffix }}_id {{ foreign_key_sql }}
+          )
+        SQL
+      end
+    end
+
+    class Enrollment{{ adapter_const_suffix }} < Granite::ORM::Base
+      primary id : Int64
+      adapter {{ adapter_literal }}
+      table_name "{{ enrollment_table }}"
+
+      belongs_to :student_{{ adapter_literal }}
+      belongs_to :klass_{{ adapter_literal }}
+
+      def self.drop_and_create
+        exec "DROP TABLE IF EXISTS {{ enrollment_table }}"
+        exec <<-SQL
+          CREATE TABLE {{ enrollment_table }} (
+            id {{ primary_key_sql }},
+            student{{ adapter_suffix }}_id {{ foreign_key_sql }},
+            klass{{ adapter_suffix }}_id {{ foreign_key_sql }}
+          )
+        SQL
+      end
+    end
+
+    class School{{ adapter_const_suffix }} < Granite::ORM::Base
+      adapter {{ adapter_literal }}
+      primary custom_id : Int64
+      field name : String
+
+      table_name "{{ school_table }}"
+
+      def self.drop_and_create
+        exec "DROP TABLE IF EXISTS {{ school_table }}"
+        exec <<-SQL
+          CREATE TABLE {{ school_table }} (
+            custom_id {{ primary_key_sql }},
+            name VARCHAR(255)
+          )
+        SQL
+      end
+    end
+
+    class Nation::County{{ adapter_const_suffix }} < Granite::ORM::Base
```

Comment:
@amberframework/granite-orm-contributors Finding an example in the "school" namespace that fit this was difficult. I'm open to suggestions if anyone has any.